firmware/python_modules/disobey2020/voltages.py

```diff
@@ -1,36 +1,44 @@
+# Analog voltage readout for the Disobey 2020 badge
 import machine
 
+# Pins
 PIN_VBAT   = 39
 PIN_VUSB   = 36
 PIN_VIDENT = 32
 
+# ADCs
 _vbat = machine.ADC(PIN_VBAT)
 _vbat.atten(machine.ADC.ATTN_11DB)
+_vbat.width(machine.ADC.WIDTH_12BIT)
 
 _vusb = machine.ADC(PIN_VUSB)
 _vusb.atten(machine.ADC.ATTN_11DB)
+_vusb.width(machine.ADC.WIDTH_12BIT)
 
 _vident = machine.ADC(PIN_VIDENT)
 _vident.atten(machine.ADC.ATTN_11DB)
 _vident.width(machine.ADC.WIDTH_12BIT)
 
+# Public functions
 def usb():
-	'''
-	Read USB input voltage
-	:return: integer, voltage in mV
-	'''
-	return int(_vusb.read()*1.97)
+    '''
+    Read USB input voltage
+    :return: integer, voltage in mV
+    '''
+    return int((_vusb.read() / 4096) * 3.9 * 2000)
```

Comment:
Where does the 3.9 come from? Is the 2k a combination of 1000 for V->mV and an attenuation?

Reply:
Oof, I should have added comments for myself...
...The 2000 is 1000 for v->mv and a times 2 because of the 100k/100k resistor divider in front of this ADC input.

Reply:
To be honest: no idea where the 3.9 comes from anymore. Note to self: check if measured voltage with multimeter matches the result of this calculation. If it does then I can use that as verification. If it doesn't it shows that my calculation here is bullshit...

Goes on my ever extending to-do list.